compiler/ilgen/IlBuilder.cpp

```diff
@@ -1154,16 +1154,17 @@ IlBuilder::setHandlerInfo(uint32_t catchType)
    }
 
 TR::Node*
-IlBuilder::genOverflowCHKTreeTop(TR::Node *operationNode)
+IlBuilder::genOverflowCHKTreeTop(TR::Node *operationNode, bool isSignedOverflow)
    {
-   TR::Node *overflowChkNode = TR::Node::createWithRoomForOneMore(TR::OverflowCHK, 3, symRefTab()->findOrCreateOverflowCheckSymbolRef(_methodSymbol), operationNode, operationNode->getFirstChild(), operationNode->getSecondChild());
+   TR::ILOpCodes overflow = isSignedOverflow? TR::OverflowCHK: TR::UnsignedOverflowCHK;
+   TR::Node *overflowChkNode = TR::Node::createWithRoomForOneMore(overflow, 3, symRefTab()->findOrCreateOverflowCheckSymbolRef(_methodSymbol), operationNode, operationNode->getFirstChild(), operationNode->getSecondChild());
    overflowChkNode->setOverflowCHKOperation(operationNode->getOpCodeValue());
    genTreeTop(overflowChkNode);
    return overflowChkNode;
    }
 
 TR::IlValue *
-IlBuilder::operationWithOverflow(TR::ILOpCodes op, TR::Node *leftNode, TR::Node *rightNode, TR::IlBuilder **handler)
+IlBuilder::operationWithOverflow(TR::ILOpCodes op, TR::Node *leftNode, TR::Node *rightNode, TR::IlBuilder **handler, bool isSignedOverflow)
```

Comment:
maybe this should be called genOverflowOperation rather than operationWithOverflow?